src/ansys/units/quantity.py

```diff
@@ -150,6 +150,9 @@ def value(self):
         """Value in contained units."""
         return self._value
 
+    # Given the complex constraints applied in the constructor, it is
+    # unclear why we are simply allowing the value to be overwritten here.
+    # See what tests are affected by removing this.
     @value.setter
     def value(self, new_value):
         self._value = new_value
```

Comment:
@seanpearsonuk We will not able to set value of Quantity as follows if we remove this.

![image](https://github.com/ansys/pyansys-units/assets/106588300/bc9659b0-efdd-4eeb-9bdd-13718deb82fe)

Reply:
Only above mentioned test is failing if we remove this. I remember that we had added this capability for float class inheritance in Quantity. Now We have removed float inheritance then we can remove this setter and related test.